dcc-release-job/dcc-release-job-fathmm/src/main/java/org/icgc/dcc/release/job/fathmm/core/FathmmPredictor.java

```diff
@@ -64,174 +67,148 @@ public int compare(final Map<String, Object> a, final Map<String, Object> b) {
-      result = newHashMap();
-      result.put("Warning", "Invalid Substitution Position");
-      return result;
+      return singletonMap("Warning", "Invalid Substitution Format");
     }
-    if (!aaChange.substring(0, 1).equals(sequenceStr.substring(substitution - 1, substitution))) {
-      result = newHashMap();
-      result.put(
-          "Warning",
-          "Inconsistent Wild-Type Residue (Expected '"
-              + sequenceStr.substring(substitution - 1, substitution) + "')");
-      return result;
-    }
-    if (left(aaChange, 1).equals(right(aaChange, 1))) {
-      result = newHashMap();
-      result.put("Warning", "Synonymous Mutation");
-      return result;
+
+    val substitutionPosition = parseSubstitutionPosition(aaChange);
+    val sequenceStr = sequence.get("sequence").toString();
+    if (substitutionPosition > sequenceStr.length()) {
+      return singletonMap("Warning", "Invalid Substitution Position");
     }
 
-    val domainList = domainQuery.bind("sequenceId", sequence.get("id")).bind("substitution", substitution).list();
+    val fromAminoAcid = aaChange.substring(0, 1);
+    val sequenceAminoAcid = sequenceStr.substring(substitutionPosition - 1, substitutionPosition);
+    if (!fromAminoAcid.equals(sequenceAminoAcid)) {
+      val description = format("Inconsistent Wild-Type Residue (Expected '%s')", sequenceAminoAcid);
 
-    for (val domain : domainList) {
-      int start = Integer.parseInt(domain.get("seq_begin").toString());
-      int end = Integer.parseInt(domain.get("seq_end").toString());
-      int hmmBegin = Integer.parseInt(domain.get("hmm_begin").toString());
-      String align = (String) domain.get("align");
-
-      String residue = mapPosition(start, end, hmmBegin, align, substitution);
-      if (null != residue) {
-        Map<String, Object> probability =
-            Iterables
-                .getFirst(probabilityQuery.bind("hmm", domain.get("hmm")).bind("residue", Integer.parseInt(residue))
-                    .list(), null);
-        if (null != probability) {
-          facade.add(probability);
-        }
-      }
+      return singletonMap("Warning", description);
     }
 
-    // Sort facade
-    if (facade.size() > 1) {
-      Collections.sort(facade, INFORMATION_COMPARATOR);
+    if (left(aaChange, 1).equals(right(aaChange, 1))) {
+      return singletonMap("Warning", "Synonymous Mutation");
     }
 
-    // TODO: Cache this!!!
-    val weightQuery = handle.createQuery("select * from \"WEIGHTS\" where id=:wid  and type='" + weights + "'");
-
-    // Phenotypes????
+    val sequenceId = (Integer) sequence.get("id");
+    val facade = getResidueProbabilities(sequenceId, substitutionPosition);
 
     // //////////////////////////////////////////////////////////////////////////////
     // Unweighted domain based prediction
     // //////////////////////////////////////////////////////////////////////////////
     for (Map<String, Object> x : facade) {
```

Comment:
I don't see any other name than `entry`, because meaning of that operation is not quite clear